### PR TITLE
Niloofar / Footer's Network Status

### PR DESCRIPTION
**Pull request by niloofar-deriv**
This PR contains the changes of network status in the footer!

Screenshot 2023-11-15 at 1 47 28 PM

### CODE REVIEW

File: .eslintrc.cjs [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = {
         'import/no-extraneous-dependencies': ['warn', { devDependencies: true }],
         'react-refresh/only-export-components': ['off', { allowConstantExport: true }],
         'react/prop-types': [2, { ignore: ['className'] }],
+        'import/no-unresolved': [2, { caseSensitive: false }],
```
Comment (balakrishna-deriv) [inline .eslintrc.cjs]: Why are we adding this eslint?

File: src/components/layout/footer/NetworkStatus.tsx [hunk 2]
```diff
@@ -1,13 +1,60 @@
...
+            window.DerivAPI = {};
+            connection?.close();
+            setStatus('offline');
+        }
+    }, [networkStatus, connection, setSocketConnection]);
+
+    const statusConfigs = {
+        online: { className: 'bg-success', tooltip: 'Online' },
+        offline: { className: 'bg-danger', tooltip: 'Offline' },
+        blinking: { className: 'animate-pulse bg-success', tooltip: 'Connecting to server' },
```
Comment (balakrishna-deriv) [inline src/components/layout/footer/NetworkStatus.tsx]: localize?

File: src/components/layout/footer/NetworkStatus.tsx [hunk 3]
```diff
@@ -1,13 +1,60 @@
...
+    const { socketConnection, setSocketConnection } = useAPIContext();
+    const connection = socketConnection.connection;
+
+    useEffect(() => {
+        if (networkStatus) {
+            /* The user is online */
+            setStatus('blinking');
+            /* Will reconnect after the timout if the network status is online and the connection is closed or closing */
+            let reconnectTimeout;
+            clearTimeout(reconnectTimeout);
```
Comment (balakrishna-deriv) [inline src/components/layout/footer/NetworkStatus.tsx]: clearTimeout should be called in the cleanup phase

File: src/components/layout/footer/NetworkStatus.tsx [hunk 4]
```diff
@@ -1,13 +1,60 @@
...
+                    setSocketConnection(newSocketConnection);
+                } else if (openState) {
+                    connection?.send({ ping: 1 }); // get stable status sooner
+                }
+            }, 500);
+
+            setStatus('online');
+        } else {
+            /* The user is offline */
+            window.DerivAPI = {};
```
Comment (balakrishna-deriv) [inline src/components/layout/footer/NetworkStatus.tsx]: why are we setting it in window?

File: src/components/layout/footer/NetworkStatus.tsx [hunk 5]
```diff
@@ -1,13 +1,60 @@
...
 
-// TODO complete the functionality + add tests
 const NetworkStatus = () => {
+    const [status, setStatus] = useState<TStatus>('online');
     const { t } = useTranslation();
+    const networkStatus = useNavigatorOnline();
+    const { socketConnection, setSocketConnection } = useAPIContext();
+    const connection = socketConnection.connection;
+
+    useEffect(() => {
```
Comment (balakrishna-deriv) [inline src/components/layout/footer/NetworkStatus.tsx]: can we abstract this effect in a separate hook(`useNetworkStatus`) to make it cleaner?

```
const networkStatus = useNetworkStatus(...);

const statusConfigs = {
     online: { className: 'bg-success', tooltip: 'Online' },
     offline: { className: 'bg-danger', tooltip: 'Offline' },
     blinking: { className: 'animate-pulse bg-success', tooltip: 'Connecting to server' },
}[status];
```

**Review by hasan-deriv — APPROVED** [review]

File: .eslintrc.cjs (hunk 1 above)
Comment (niloofar-deriv) [reply .eslintrc.cjs]: I had an issue with committing my changes that's why I added this rule, but I guess I can remove it so will try that

File: src/components/layout/footer/NetworkStatus.tsx (hunk 2 above)
Comment (niloofar-deriv) [reply src/components/layout/footer/NetworkStatus.tsx]: I'm passing this here: 

```
content={t('Network status: {{tooltip}}', { tooltip })}
```

File: src/components/layout/footer/NetworkStatus.tsx (hunk 5 above)
Comment (niloofar-deriv) [reply src/components/layout/footer/NetworkStatus.tsx]: Yes sure

File: src/components/layout/footer/NetworkStatus.tsx (hunk 4 above)
Comment (niloofar-deriv) [reply src/components/layout/footer/NetworkStatus.tsx]: As we talked we are using this pattern in the app.deriv project too.

File: src/components/layout/footer/NetworkStatus.tsx (hunk 3 above)
Comment (niloofar-deriv) [reply src/components/layout/footer/NetworkStatus.tsx]: Sure will change this 🙏

**Review by balakrishna-deriv — APPROVED** [review]

**Pull request merged**